ytfzf

```diff
@@ -1062,6 +1062,13 @@ preview_display_image () {
 			dep_check "display" || die 3 "\nimagemagick is not installed\n"
 			killall display
 			display "$thumb_dir/${id}.jpg" ;;
+		kitty)
+			printf '\n'
+			[ "$TERM" = "xterm-kitty" ] || die 3 "\nnot running inside kitty\n"
```

Comment:
This is not a good way to check if it's running in kitty, because $TERM is just what the user puts in `.bashrc` not necessarily the terminal they're running

Reply:
You could try something like
```sh
kitty +icat --detect-support --detection-timeout=4 || die 3 "\nnot running inside kitty\n"
```
and also put that after the `dep_check kitty`
You can also change detection-timeout to whatever you think is good (the default is 10 seconds which seems long)

Reply:
Yes, any use of --detect-support or even --detection-timeout seem to break things.

I've tried to do other things like `kitty +kitten query_terminal --wait-for=2 name` but that /also/ breaks it. I think kitty misdetect things (thus the need for transfer-mode) and about anything will break it.

Don't really have any better ideas right now, TERM was just a basic check in case "oops happened to start it in the wrong terminal" so things don't get crazy (not that you can't just ctrl+c out of it but well).

Reply:
...but well, guess users may also want to set TERM to something else and while keeping this working, so removing the TERM check may be better. It'd just rely on users using the right settings.

Reply:
aight here's a hacky solution
in `preview_start` we can add a case for kitty
```sh
kitty)
    kitty +kitten query_terminal name &
    pid="$!"
    sleep 1
    kill "$pid" && die 3 "not using kitty terminal"
```

we can change `sleep 1` to be shorter but we can't make it too short.

---

This works because when the terminal isn't kitty, `query_terminal` takes AGES, so if we try killing it before it's done kill will be sucessful meaning we're not in kitty, if kill is successful we are in kitty.

I feel like we could do better than this though.

Reply:
> It'd just rely on users using the right settings.

This is how i'd do it too.